---
title: "Get started"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{getstarted}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

#### Simple tutorial using default bayesian method on example file

If not already done, download the archive containing the reference databases and the bayesian models from `zenodo.org`, using the `inborutils` package. You can change the `path` option to where you want to download the archive (default is current directory '.'):

```
remotes::install_github("inbo/inborutils")
inborutils::download_zenodo("10.5281/zenodo.13733183", path=".")
```

Store the path to the archive containing the reference databases and the bayesian models:

```
refdata_archive_path = "path/to/genomesizeRdata.tar.gz"
```

Read the example input file from the package. This example data is a subset of the dataset from [Labouyrie et al. 2023](https://doi.org/10.1038/s41467-023-37937-4):


```
example_input_file = system.file("extdata", "example_input.csv", package = "genomesizeR")
```

Load the package:

```
library(genomesizeR)
```

Run the main function to get the estimated genome sizes (with the default method which is the bayesian method):
First, the genome sizes are predicted from the taxa:

```
  results = estimate_genome_size(example_input_file, sep='\t', match_column='TAXID', output_format='input', ci_threshold = 0.3)
  
  #############################################################################
  # Genome size estimation summary:
  #
  #  22.22222 % estimations achieving required precision
  #
       Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
    3007721   5408472  16980834  23969767  41811396 143278734 
  
  # Estimation status:
  Confidence interval to estimated size ratio > ci_threshold                                                         OK 
                                                         140                                                         40 
```

#### Plot genome size histogram per sample

Then, the results can be visualized using the plotting functions provided. This histogram shows the estimated genome sizes for each sample. 

```{r, echo = FALSE, warning=F, message=F}
  library(genomesizeR)
  example_output_data = system.file("extdata", "results_bayesian.csv", package = "genomesizeR")
  refdata_archive_path = system.file("extdata", "genomesizeRdata.tar.gz", package = "genomesizeR")
  results = read.csv2(example_output_data, sep=',', dec = '.')
```

```{r, echo = TRUE}
  plotted_df = plot_genome_size_histogram(results)
```

#### Plot genome size histogram for one sample

```{r, echo = TRUE}
  plotted_df = plot_genome_size_histogram(results, only_sample='16S_1')
```

#### Plot genome size boxplot per sample

This boxplot shows the estimated genome sizes for each sample:

```{r, echo = TRUE}
  plotted_df = plot_genome_size_boxplot(results)
```

#### Plot genome size boxplot for one sample

```{r, echo = TRUE}
  plotted_df = plot_genome_size_boxplot(results, only_sample='ITS_1')
```

#### Plot simplified taxonomic tree with colour-coded estimated genome sizes

This tree shows the taxonomic relationships as well as the estimated genome sizes. The difference between the genome size distribution of bacteria (16S marker) and fungi (ITS marker) is visible.

```{r, echo = TRUE, fig.width= 22, fig.height= 22, warning=F, message=F}
  plotted_df = plot_genome_size_tree(results, refdata_archive_path)
```
