<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Get started • genomesizeR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get started">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">genomesizeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0.0001</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/genomesizeR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/compare_methods.html">Comparing estimation methods</a></li>
    <li><a class="dropdown-item" href="../articles/input_output.html">About the input and the output</a></li>
    <li><a class="dropdown-item" href="../articles/methods.html">Method description</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get started</h1>
            
      

      <div class="d-none name"><code>genomesizeR.Rmd</code></div>
    </div>

    
    
<div class="section level4">
<h4 id="simple-tutorial-using-default-bayesian-method-on-example-file">Simple tutorial using default bayesian method on example file<a class="anchor" aria-label="anchor" href="#simple-tutorial-using-default-bayesian-method-on-example-file"></a>
</h4>
<p>If not already done, download the archive containing the reference
databases and the bayesian models from <code>zenodo.org</code>, using
the <code>inborutils</code> package. You can change the
<code>path</code> option to where you want to download the archive
(default is current directory ‘.’):</p>
<pre><code><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"inbo/inborutils"</span><span class="op">)</span></span>
<span><span class="fu">inborutils</span><span class="fu">::</span><span class="fu">download_zenodo</span><span class="op">(</span><span class="st">"10.5281/zenodo.13733183"</span>, path<span class="op">=</span><span class="st">"."</span><span class="op">)</span></span></code></pre>
<p>Store the path to the archive containing the reference databases and
the bayesian models:</p>
<pre><code><span><span class="va">refdata_archive_path</span> <span class="op">=</span> <span class="st">"path/to/genomesizeRdata.tar.gz"</span></span></code></pre>
<p>Read the example input file from the package. This example data is a
subset of the dataset from <a href="https://doi.org/10.1038/s41467-023-37937-4" class="external-link">Labouyrie et
al. 2023</a>:</p>
<pre><code><span><span class="va">example_input_file</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example_input.csv"</span>, package <span class="op">=</span> <span class="st">"genomesizeR"</span><span class="op">)</span></span></code></pre>
<p>Load the package:</p>
<pre><code><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://scionresearch.github.io/genomesizeR/">genomesizeR</a></span><span class="op">)</span></span></code></pre>
<p>Run the main function to get the estimated genome sizes (with the
default method which is the bayesian method): First, the genome sizes
are predicted from the taxa:</p>
<pre><code>  results = estimate_genome_size(example_input_file, sep='\t', match_column='TAXID', output_format='input', ci_threshold = 0.3)
  
  #############################################################################
  # Genome size estimation summary:
  #
  #  22.22222 % estimations achieving required precision
  #
       Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
    3007721   5408472  16980834  23969767  41811396 143278734 
  
  # Estimation status:
  Confidence interval to estimated size ratio &gt; ci_threshold                                                         OK 
                                                         140                                                         40 </code></pre>
</div>
<div class="section level4">
<h4 id="plot-genome-size-histogram-per-sample">Plot genome size histogram per sample<a class="anchor" aria-label="anchor" href="#plot-genome-size-histogram-per-sample"></a>
</h4>
<p>Then, the results can be visualized using the plotting functions
provided. This histogram shows the estimated genome sizes for each
sample.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="va">plotted_df</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_genome_size_histogram.html">plot_genome_size_histogram</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span></code></pre></div>
<p><img src="genomesizeR_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="plot-genome-size-histogram-for-one-sample">Plot genome size histogram for one sample<a class="anchor" aria-label="anchor" href="#plot-genome-size-histogram-for-one-sample"></a>
</h4>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="va">plotted_df</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_genome_size_histogram.html">plot_genome_size_histogram</a></span><span class="op">(</span><span class="va">results</span>, only_sample<span class="op">=</span><span class="st">'16S_1'</span><span class="op">)</span></span></code></pre></div>
<p><img src="genomesizeR_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="plot-genome-size-boxplot-per-sample">Plot genome size boxplot per sample<a class="anchor" aria-label="anchor" href="#plot-genome-size-boxplot-per-sample"></a>
</h4>
<p>This boxplot shows the estimated genome sizes for each sample:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="va">plotted_df</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_genome_size_boxplot.html">plot_genome_size_boxplot</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span></code></pre></div>
<p><img src="genomesizeR_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="plot-genome-size-boxplot-for-one-sample">Plot genome size boxplot for one sample<a class="anchor" aria-label="anchor" href="#plot-genome-size-boxplot-for-one-sample"></a>
</h4>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="va">plotted_df</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_genome_size_boxplot.html">plot_genome_size_boxplot</a></span><span class="op">(</span><span class="va">results</span>, only_sample<span class="op">=</span><span class="st">'ITS_1'</span><span class="op">)</span></span></code></pre></div>
<p><img src="genomesizeR_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="plot-simplified-taxonomic-tree-with-colour-coded-estimated-genome-sizes">Plot simplified taxonomic tree with colour-coded estimated genome
sizes<a class="anchor" aria-label="anchor" href="#plot-simplified-taxonomic-tree-with-colour-coded-estimated-genome-sizes"></a>
</h4>
<p>This tree shows the taxonomic relationships as well as the estimated
genome sizes. The difference between the genome size distribution of
bacteria (16S marker) and fungi (ITS marker) is visible.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="va">plotted_df</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_genome_size_tree.html">plot_genome_size_tree</a></span><span class="op">(</span><span class="va">results</span>, <span class="va">refdata_archive_path</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Untarring reference data</span></span>
<span><span class="co">## Using reference data in: /tmp/RtmpVTRxZC/refdata</span></span>
<span><span class="co">## Untarring taxonomy</span></span>
<span><span class="co">## Using taxonomy: /tmp/RtmpVTRxZC/taxdump</span></span></code></pre>
<p><img src="genomesizeR_files/figure-html/unnamed-chunk-6-1.png" width="2112"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Celine Mercier, Joane Ellouet.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
